### PR TITLE
Updated samples to use the submit range from the FrameGraphExecuteContext

### PR DESCRIPTION
**Pull request by dmcdiarmid-ly**
Companion PR to O3DE #10504

Signed-off-by: dmcdiarmid-ly <63674186+dmcdiarmid-ly@users.noreply.github.com>

### CODE REVIEW

File: Gem/Code/Source/RHI/SubpassExampleComponent.cpp [hunk 1]
```diff
@@ -186,16 +186,29 @@ namespace AtomSampleViewer
...
+        //
+        //    uint32_t meshCount = 0;
+        //    for (uint32_t i = 0; i < modelDataList.size(); ++i)
+        //    {
+        //        modelDataList[i].m_modelType = types[i];
+        //        meshCount += m_models[modelDataList[i].m_modelType]->GetLods()[0]->GetMeshes().size();
+        //    }
+        //};
+        //
+        //setModelType(AZStd::span<const ModelType>(&opaqueModels[0], AZ_ARRAY_SIZE(opaqueModels)), m_opaqueModelsData);
```
Comment (antonmic) [inline Gem/Code/Source/RHI/SubpassExampleComponent.cpp:201]: Remove if unused, or add TODO

**Review by antonmic — APPROVED** [review]

File: Gem/Code/Source/RHI/SubpassExampleComponent.cpp (hunk 1 above)
Comment (dmcdiarmid-ly) [reply Gem/Code/Source/RHI/SubpassExampleComponent.cpp:201]: Removed.

File: Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp [hunk 2]
```diff
@@ -259,7 +259,7 @@ namespace AtomSampleViewer
                 frameGraph.UseShaderAttachment(attachmentDescriptor, RHI::ScopeAttachmentAccess::ReadWrite);
             }
 
-            frameGraph.SetEstimatedItemCount(1);
+            frameGraph.SetEstimatedItemCount(2);
```
Comment (moudgils) [inline Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp:262]: Why does the the ItemCount need to change here?

File: Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp [hunk 3]
```diff
@@ -358,7 +360,7 @@ namespace AtomSampleViewer
             }
 
             // We will submit a single draw item.
-            frameGraph.SetEstimatedItemCount(1);
+            frameGraph.SetEstimatedItemCount(2);
```
Comment (moudgils) [inline Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp:363]: Same

File: Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp (hunk 2 above)
Comment (dmcdiarmid-ly) [reply Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp:262]: It's doing multiple submits.  The previous item count of 1 was therefore incorrect.

File: Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp (hunk 3 above)
Comment (dmcdiarmid-ly) [reply Gem/Code/Source/RHI/InputAssemblyExampleComponent.cpp:363]: Same as above

**Review by moudgils — APPROVED** [review]

**Pull request merged**